packages/plugin-hardhat/src/deploy-proxy.ts

```diff
@@ -20,17 +20,19 @@ import {
   getInitializerData,
 } from './utils';
 
+type InstanceOf<F extends ContractFactory> = ReturnType<F['attach']>;
```

Comment:
Can you call this `InstanceOfFactory` or something with Factory?

Reply:
yeah, good call.